app/controllers/pages_controller.rb

```diff
@@ -1,8 +1,65 @@
 class PagesController < Comfy::Admin::Cms::PagesController
-  before_action :build_file,     only: [:new, :edit]
-  before_action :set_categories, only: [:new, :edit, :create]
-  before_action :set_pages,      only: :index
-  before_action :set_activity_log, only: [:new, :edit, :create]
+
+  helper 'page_blocks'
+
+  before_action :check_permissions
+  before_action :check_alternate_available, only: [:edit, :update, :destroy]
+  before_action :check_can_destroy, only: :destroy
+
+  def index
+    @all_pages = Comfy::Cms::Page.includes(:layout, :site, :categories)
+    @pages_by_parent = pages_grouped_by_parent
+    @pages = apply_filters
+  end
+
+  def new
```

Comment:
As discussed offline, I include an empty method to make it easy to see that this action does get used without having to refer to routes/views, plus also with this controller inheriting from another controller you don't need to go look in the parent to check the exact behaviour.